### PR TITLE
[SPARK-15526][ML][FOLLOWUP][test-maven] Make JPMML provided scope to avoid including unshaded JARs, and repromote to compile in MLlib

### PR DESCRIPTION
**Pull request by srowen**
## What changes were proposed in this pull request?

Following the comment at https://issues.apache.org/jira/browse/SPARK-15526?focusedCommentId=16086106&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-16086106 -- this change actually needed a little more work to be complete.

This also marks JPMML as `provided` to make sure its JARs aren't included in the `jars` output, but then scopes to `compile` in `mllib`. This is how Guava is handled.

## How was this patch tested?

Checked result in `assembly/target/scala-2.11/jars` to verify there are no JPMML jars. Maven and SBT builds still work.

### CODE REVIEW

**Comment by srowen** [comment]
CC @vanzin

**Comment by vanzin** [comment]
Can you add `[test-maven]` to the title so that the PR builder runs the maven build?

**Comment by vanzin** [comment]
retest this please

File: pom.xml [hunk 1]
```diff
@@ -2511,8 +2523,8 @@
   <profiles>
 
     <!--
-      This profile is enabled automatically by the sbt built. It changes the scope for the guava
-      dependency, since we don't shade it in the artifacts generated by the sbt build.
+      This profile is enabled automatically by the sbt build. It changes the scope for shaded
```
Comment (vanzin) [inline pom.xml]: I wonder if this profile is even necessary, given it does not include jetty which is also "provided" by default... but I'm too lazy to do history spelunking at the moment.

**Review by vanzin — COMMENTED** [review]
LGTM pending maven tests. If those fail you may need to mirror what `core/` does with these dependencies (see `copy-dependencies` invocation there, and look for `core/target/jars` in `AbstractCommandBuilder.java`.

BTW those changes might be nice to have in any case, since they allow `SPARK_PREPEND_CLASSES` to work.

File: pom.xml (hunk 1 above)
Comment (srowen) [inline pom.xml]: Likewise, just updating this to update it

**Review by srowen — COMMENTED** [review]
Yes, I added the `copy-dependencies` element to `mllib`, and adapted it, I believe. @vanzin

File: mllib/pom.xml [hunk 2]
```diff
@@ -139,8 +133,38 @@
...
+
   <build>
     <outputDirectory>target/scala-${scala.binary.version}/classes</outputDirectory>
     <testOutputDirectory>target/scala-${scala.binary.version}/test-classes</testOutputDirectory>
+    <plugins>
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-dependency-plugin</artifactId>
+        <executions>
+          <!-- When using SPARK_PREPEND_CLASSES Spark classes compiled locally don't use
```
Comment (vanzin) [inline mllib/pom.xml]: You need to add code in `AbstractCommandBuilder.java` (method `buildClassPath`) for this to have an actual effect.

Comment (srowen) [inline mllib/pom.xml]: Thanks @vanzin, like this? https://github.com/apache/spark/pull/18637/commits/9c3ab057ad1ff89ab726ea86774692ef22151b49

**Comment by vanzin** [comment]
LGTM. Merging to master.

**Pull request closed without merging**